cooltools/api/eigdecomp.py

```diff
@@ -138,15 +98,191 @@ def cis_eig(
+        'MAD_explained' - amount of Median Absolute Deviation from the median of
+                          `eigvecs` explained by `phasing_track`
+                          (i.e. COMED(eigvec, phasing_track) * MAD(eigvec)).
+    map : callable, optional
+        Map functor for parallelization.
+
+    Returns
+    -------
+    eigvals, eigvecs : DataFrame
+        Dataframes with eigenvalues for each region and a bin table of
+        eigenvectors.
+
+    Notes
+    -----
+    The amplitude of each eigenvector is weighted by its corresponding
+    eigenvalue: ``sqrt(abs(eigval))``.
+
+    ALWAYS check your EVs by eye. The first one occasionally does not reflect
+    the compartment structure, but instead describes chromosomal arms or
+    translocation blowouts. Possible mitigations: employ `view_df` (e.g. arms)
+    to avoid issues with chromosomal arms, consider blacklisting regions with
+    translocations during balancing.
+
+    The ``reorder`` option is designed to report the most "biologically"
+    informative eigenvectors first, and prevent eigenvector swapping caused by
+    translocations. In reality, however, sometimes it shows poor performance and
+    may lead to reporting of non-informative eigenvectors.
+
+    """
+    # Determine chromosomes from cooler, if view_df not specified
+    if view_df is None:
+        view_df = make_cooler_view(clr)
+    else:
+        # Make sure view_df is a proper viewframe
+        try:
+            _ = is_compatible_viewframe(
+                view_df,
+                clr,
+                check_sorting=True,
+                raise_errors=True,
+            )
+        except Exception as e:
+            raise ValueError("view_df is not a valid viewframe or incompatible") from e
+
+    # Check if cooler is balanced
+    try:
+        _ = is_cooler_balanced(clr, clr_weight_name, raise_errors=True)
+    except Exception as e:
+        raise ValueError(
+            f"provided cooler is not balanced or {clr_weight_name} is missing"
+        ) from e
+
+    # Ignore diags as in cooler unless specified
+    ignore_diags = (
+        clr._load_attrs(f"bins/{clr_weight_name}").get("ignore_diags", 2)
+        if ignore_diags is None
+        else ignore_diags
+    )
 
-    # Orient and reorder
+    bins = clr.bins()[:]
+
+    # Adjust phasing track as necessary
     if phasing_track is not None:
-        eigvals, eigvecs = _phase_eigs(eigvals, eigvecs, phasing_track, sort_metric)
+        phasing_track = align_track_with_cooler(
+            phasing_track,
+            clr,
+            view_df=view_df,
+            clr_weight_name=clr_weight_name,
+            mask_clr_bad_bins=True,
+            drop_track_na=True # this adds check for chromosomes that have all missing values
+        )
+
+    # Prepare output table for eigen vectors
+    eigvec_table = bioframe.assign_view(bins, view_df).dropna(subset=["view_region"], axis=0)
+    eigvec_table = eigvec_table.loc[:, bins.columns]
```

Comment:
can use one of the pandas methods to add several empty columns at once - as in [here](https://sparkbyexamples.com/pandas/pandas-add-an-empty-column-to-dataframe/) : `assign, reindex` ...

e.g.
```python
eigvec_table = eigvec_table.reindex(columns=bins.columns+eigvec_columns)
```
this should pre-fill `eigvec_columns` with NaNs

Reply:
also a great place to get rid of "weights" column here - as per #394 ...

Reply:
nice!

Reply:
https://github.com/nvictus/cooltools/pull/2